docker/studyu-cli.sh

```diff
@@ -0,0 +1,479 @@
+#!/bin/bash
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+PARENT_DIR=$(dirname "$SCRIPT_DIR")
+
+# Color codes for better output
+green='\e[0;32m'
+yellow='\e[1;33m'
+red='\e[0;31m'
+bold='\e[1m'
+reset='\e[0m'
+
+# Configuration file path
+CONFIG_FILE="$SCRIPT_DIR/.studyu_config"
+
+# Dictionary to map components to Docker Compose files
+declare -A compose_files=(
+  ["supabase-db"]="supabase/docker-compose-db.yml"
+  ["supabase"]="supabase/docker-compose.yml"
+  ["proxy"]="nginx/docker-compose-proxy.yml"
+  # ["studyu"]="studyu/docker-compose.yml"
+  ["studyu-app"]="studyu/docker-compose-app.yml"
+  ["studyu-designer"]="studyu/docker-compose-designer.yml"
+)
+available_components="${bold}Available components:\n${reset}"
+available_components+="  supabase-db        Postgres database used as Supabase backend\n"
+available_components+="  supabase           Supabase middleware (needs supabase-db)\n"
+available_components+="  proxy              Nginx reverse proxy to access the Supabase frontend if no StudyU component is started\n"
+# available_components+="  studyu             StudyU App and StudyU Designer (includes proxy)\n"
+available_components+="  studyu-app         StudyU App (includes proxy)\n"
+available_components+="  studyu-designer    StudyU Designer (includes proxy)"
+
+# Function to read configuration values from the file
+read_config() {
+  local failOnError="${1:-true}"
+
+  if [ "$failOnError" = true ] && [ ! -f "$CONFIG_FILE" ]; then
+    echo -e "${red}Error: Configuration file not found. Run 'config' to create one.${reset}"
+    exit 1
+  fi
+
+  # Read the configuration file
+  set -a
+  #shellcheck source=.studyu_config
+  source "$CONFIG_FILE"
+  set +a
+
+  # Check if the required variables are present
+  if [ "$failOnError" = true ] && { [ -z "$STUDYU_DEFAULT_COMPONENTS" ] || [ -z "$STUDYU_PATH" ] || [ -z "$STUDYU_BASE_BRANCH" ]; }; then
+    echo -e "${red}Error: Configuration file $CONFIG_FILE is missing some entries.${reset}"
+    echo "Please run 'config' to set the required options."
+    exit 1
+  fi
+
+  cd "$STUDYU_PATH/docker/" || exit 1
```

Comment:
Oh and another thing: why is there a need to set STUDYU_PATH manually at all? The script's parent dir is always correct, isn't it? Moving the script would break many things anyways.

Reply:
I thought of adding the possibility in the future to install the script to a PATH dir in order to make it run globally as a `studyu` command. After setting the STUDYU_PATH this might be possible.

Reply:
Okay, that makes sense. But then we also (1) need to move the config to something like `~/.studyu_config` or (2) rely on `STUDYU_PATH` already being set ahead of time.

Reply:
Since the script should be usable by multiple users on the same machine, relying on home directories is no good. However, we can rely on `STUDYU_PATH` being set, since it is part of the configuration file `.studyu_config` that is generated at first launch.

```
❯ cat .studyu_config
STUDYU_DEFAULT_COMPONENTS="supabase-db supabase studyu-app studyu-designer"
STUDYU_PATH="/home/Johannes/studyu"
STUDYU_BASE_BRANCH="dev"
```

Without `STUDYU_PATH` being set, no action can be performed, since this check inside `read_config` is executed before running any command:

https://github.com/hpi-studyu/studyu/blob/3692ccdb767580eb6f579730f5e92f59023b93b4/docker/studyu-cli.sh#L49-L53

When running `config` the first time, the configuration file will be created at (a more or less hardcoded) path.

https://github.com/hpi-studyu/studyu/blob/3692ccdb767580eb6f579730f5e92f59023b93b4/docker/studyu-cli.sh#L14

If someone would like to move `studyu-cli.sh` to a directory that is not writable by the user, setting the `STUDYU_CONFIG_FILE` env variable and recreating the config or move the old one to the new position should be sufficient. I added this:

```
# Configuration file path
# Check if the STUDYU_CONFIG_FILE environment variable is set
if [ -n "$STUDYU_CONFIG_FILE" ]; then
    CONFIG_FILE="$STUDYU_CONFIG_FILE"
else
    # Otherwise, fallback to the hardcoded path
    SCRIPT_DIR="$(dirname "$0")"
    CONFIG_FILE="$SCRIPT_DIR/.studyu_config"
fi
```